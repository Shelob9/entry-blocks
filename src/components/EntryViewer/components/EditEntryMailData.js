import {RenderGroup} from '@caldera-labs/components';
import React, {Fragment} from 'react';
const ID = 'cf-mail-headers-edit';


function createConfigFields(mailData, onChange) {
	const {
		recipients,
		subject,
		replyto,
		from_name,
		html
	} = mailData;


	return [
		{
			'id': ID + '-from',
			'label': 'From (reply-to)',
			'desc': 'Reply-to email address',
			'type': 'email',
			'description': false,
			value: replyto,
			onValueChange: (from) => {
				onChange({
					...mailData,
					from
				})
			}

		},
		{
			'id': ID + '-from_name',
			'label': 'Name email is sent from',
			'desc': 'Comma separated list of recipients.',
			'type': 'text',
			'description': false,
			value: from_name,
			onValueChange: (from_name) => {
				onChange({
					...mailData,
					from_name
				})
			}

		},
		{
			'id': ID + '-recipients',
			'label': 'Recipients',
			'desc': 'Comma separated list of recipients.',
			'type': 'text',
			'description': false,
			value: recipients,
			onValueChange: (recipients) => {
				onChange({
					...mailData,
					recipients
				})
			}
		},
		{
			'id': ID + '-subject',
			'label': 'Message Subject',
			'type': 'text',
			'description': false,
			value: subject,
			onValueChange: (subject) => {
				onChange({
					...mailData,
					subject
				})
			}

		},
		{
			'id': ID + '-html',
			'type': 'dropdown',
			'label': 'Content type',
			'description': 'Choose content type, default is HTML',
			options: [
				{
					label: 'HTML',
					value: 'false'
				},
				{
					label: 'Plain Text',
					value: 'true'
				}
			],
			value: html,
			onValueChange: (html) => {
				onChange({
					...mailData,
					html
				})
			}
		}
	];


}


export const EditEntryMailData = (
	{ mailData,
	  onChange,
}) => {


	return (
		<Fragment>
			<RenderGroup
				configFields={createConfigFields(mailData, onChange)}
				className={'cf-mail-headers-edit'}
			/>

		</Fragment>
	)
};


