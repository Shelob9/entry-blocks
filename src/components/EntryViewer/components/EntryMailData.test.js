import renderer from "react-test-renderer";
import React from "react";
import {EditEntryMailData} from "./EditEntryMailData";
import Enzyme, {shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

describe( 'EditEntryMailData component ', () => {

	const onChange =jest.fn();
	const mailData=  {
		recipients: 'roy@hiroy.club',
		subject: 'Hi Roy',
		replyto: 'no-reply@calderawp.com',
		from_name: 'Roy Sivan',
		html: '<p>Hi Roy</p>'
	};
	test('EntryMailDatau matches snapshot', () => {
		expect(
			renderer.create(
				<EditEntryMailData
					mailData={mailData}
					onChange={onChange}

				/>
			)
		).toMatchSnapshot();
	});


	it( 'Sends updates', () => {
		const onChange = jest.fn();
		const component = mount( <EditEntryMailData
			mailData={mailData}
			onChange={onChange}

		/>);
		component.find( 'select' ).simulate( 'change', {target: {value: 'false'}});
		expect(onChange.mock.calls.length ).toBe(1);
		expect(onChange).toMatchSnapshot();
	});
});
