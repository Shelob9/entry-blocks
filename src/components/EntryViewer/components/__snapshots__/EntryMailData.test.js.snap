exports[`EditEntryMailData component  EntryMailDatau matches snapshot 1`] = `
<div
  className="cf-mail-headers-edit">
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="cf-mail-headers-edit-from">
        From (reply-to)
      </label>
      <input
        aria-describedby={null}
        className="field-config field-config"
        disabled={false}
        id="cf-mail-headers-edit-from"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        required={false}
        type="email"
        value="no-reply@calderawp.com" />
    </div>
  </div>
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="cf-mail-headers-edit-from_name">
        Name email is sent from
      </label>
      <input
        aria-describedby={null}
        className="field-config field-config"
        disabled={false}
        id="cf-mail-headers-edit-from_name"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        required={false}
        type="text"
        value="Roy Sivan" />
    </div>
  </div>
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="cf-mail-headers-edit-recipients">
        Recipients
      </label>
      <input
        aria-describedby={null}
        className="field-config field-config"
        disabled={false}
        id="cf-mail-headers-edit-recipients"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        required={false}
        type="text"
        value="roy@hiroy.club" />
    </div>
  </div>
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="cf-mail-headers-edit-subject">
        Message Subject
      </label>
      <input
        aria-describedby={null}
        className="field-config field-config"
        disabled={false}
        id="cf-mail-headers-edit-subject"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        required={false}
        type="text"
        value="Hi Roy" />
    </div>
  </div>
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="cf-mail-headers-edit-html">
        Content type
      </label>
      <select
        className="field-config field-config"
        disabled={false}
        id="cf-mail-headers-edit-html"
        onChange={[Function]}
        value="<p>Hi Roy</p>">
        <option
          className="caldera-config-option"
          value="false">
          HTML
        </option>
        <option
          className="caldera-config-option"
          value="true">
          Plain Text
        </option>
      </select>
      <p
        className="description"
        id="cf-mail-headers-edit-html-description">
        Choose content type, default is HTML
      </p>
    </div>
  </div>
</div>
`;

exports[`EditEntryMailData component  Sends updates 1`] = `[Function]`;
